lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,416 @@
+use ensogl_text::formatting;
+
+
+// ==============
+// === Export ===
+// ==============
+
+pub mod model;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// Base of the exponentiation for precision increment steps of 10^x
+pub const PRECISION_STEP_BASE: f32 = 10.0;
+
+
+
+// ========================
+// === Slider component ===
+// ========================
+
+/// Slider component structure
+pub struct Slider {
+    pub frp: Frp,
+    model:   Rc<Model>,
+    pub app: Application,
+}
+
+impl Slider {
+    /// Constructor
+    pub fn new(app: &Application) -> Self {
+        let model = Rc::new(Model::new(app));
+        let app = app.clone_ref();
+        let frp = Frp::new();
+
+        Self { frp, model, app }.init()
+    }
+}
+
+impl display::Object for Slider {
+    fn display_object(&self) -> &display::object::Instance {
+        self.model.display_object()
+    }
+}
+
+impl Deref for Slider {
+    type Target = Frp;
+    fn deref(&self) -> &Self::Target {
+        &self.frp
+    }
+}
+
+impl FrpNetworkProvider for Slider {
+    fn network(&self) -> &enso_frp::Network {
+        self.frp.network()
+    }
+}
+
+impl application::View for Slider {
+    fn label() -> &'static str {
+        "Slider"
+    }
+
+    fn new(app: &Application) -> Self {
+        Self::new(app)
+    }
+
+    fn app(&self) -> &Application {
+        &self.app
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+ensogl_core::define_endpoints_2! {
+    Input {
+        /// Set the width of the component
+        set_width(f32),
+        /// Set the height of the component
+        set_height(f32),
+
+        /// Set the color of the slider
+        set_slider_color(color::Lcha),
+        /// Set the color of the background
+        set_background_color(color::Lcha),
+
+        /// Set the slider value
+        set_value(f32),
+        /// Set the default value to reset a slider to
+        set_value_default(f32),
+        /// Set the minimum to which the value is clamped
+        set_value_min(f32),
+        /// Set the maximum to which the value is clamped
+        set_value_max(f32),
+        /// Set the color of the value text display
+        set_value_color(color::Lcha),
+
+        /// Set the default precision at which the slider operates
+        set_precision(f32),
+        /// Set the margin above/below the slider beyond which the precision is adjusted up/downwards
+        set_precision_step_margin(f32),
+        /// Set the distance of vertical mouse movement needed to increment/decrement the precision to the next step
+        set_precision_step_size(f32),
+
+        /// Set a slider label
+        set_label(ImString),
+        /// Set the color of the slider label
+        set_label_color(color::Lcha),
+        /// Set whether the label is displayed
+        set_label_visible(bool),
+        /// Set whether the label is shown inside the slider, as opposed to left of it
+        set_label_inside(bool),
```

Comment:
`set_label_inside(false)` is IMO a bad API. It's bad because its not explicit. It should be `setlabel_placement(LabelPlacement)`. Please think about other names to make them more explicit.